### PR TITLE
Php 5.6

### PR DESCRIPTION
**Pull request by dbu**
* dropping php 5.5 support
* convert to short array notation
* use createMock
* use ::class pseudo-constant

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
When having such a nice list can you create checkboxes next time to visualize the state of this PR, please.

File: .travis.yml [hunk 1]
```diff
@@ -35,10 +33,11 @@ before_install:
   - composer self-update
   - if [ "$DEPS" = "dev" ]; then perl -pi -e 's/^}$/,"minimum-stability":"dev"}/' composer.json; fi
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require symfony/symfony:${SYMFONY_VERSION} --no-update; fi
+  - phpunit --version
 
 install: composer update --prefer-dist $COMPOSER_FLAGS
 
-script: vendor/bin/simple-phpunit
```
Comment (dbu) [inline .travis.yml]: simple-phpunit installs phpunit 5.3

File: composer.json [hunk 2]
```diff
@@ -12,11 +12,12 @@
...
     ],
     "require": {
-        "php": "^5.5.9|^7.0",
+        "php": "^5.6.0|^7.0",
         "symfony/framework-bundle": "^2.8|^3.0"
     },
     "require-dev": {
         "symfony/security-bundle": "^2.8|^3.0",
+        "phpunit/phpunit": "5.6.*",
         "symfony/phpunit-bridge": "^3.2",
```
Comment (dbu) [inline composer.json]: wouter, do we still need the bridge (for deprecations or such?) or can we remove this when we can use phpunit 5.6 for everything?

File: .travis.yml (hunk 1 above)
Comment (wouterj) [inline .travis.yml]: Please revert this change. We use `simple-phpunit` to avoid building PHPunit with Yaml.

You can change the version that's used by simple-phpunit by setting the `SYMFONY_PHPUNIT_VERSION` env var. Please set that one to 5.6 (or 5.7) instead.

Comment (dbu) [inline .travis.yml]: ah, thats how we could control that, thanks.

but it seems that in versions  5.6 that we get, the yaml issue is fixed. or is that a problem that could come up again in the future?

Comment (wouterj) [inline .travis.yml]: > or is that a problem that could come up again in the future?

Yep, e.g. when Symfony 4 will be released. PHPunit 5.6 requries symfony/yaml 3, so it'll fail to install or use an incompatible Yaml component version.

**Comment by dbu** [comment]
@wouterj happy with this? it seems good to me

File: src/DependencyInjection/Compiler/DoctrineOrmMappingsPass.php [hunk 3]
```diff
@@ -107,11 +107,11 @@ protected function getChainDriverServiceName(ContainerBuilder $container)
...
      *                                       optional
      */
-    public static function createXmlMappingDriver(array $mappings, array $managerParameters = array(), $enabledParameter = false)
+    public static function createXmlMappingDriver(array $mappings, array $managerParameters = [], $enabledParameter = false)
     {
-        $arguments = array($mappings, '.orm.xml');
+        $arguments = [$mappings, '.orm.xml'];
         $locator = new Definition('Doctrine\Common\Persistence\Mapping\Driver\SymfonyFileLocator', $arguments);
-        $driver = new Definition('Doctrine\ORM\Mapping\Driver\XmlDriver', array($locator));
+        $driver = new Definition('Doctrine\ORM\Mapping\Driver\XmlDriver', [$locator]);
```
Comment (ElectricMaxxx) [inline src/DependencyInjection/Compiler/DoctrineOrmMappingsPass.php]: can we use `::class` here?

Comment (dbu) [inline src/DependencyInjection/Compiler/DoctrineOrmMappingsPass.php]: you are right. but actually we don't need that class anymore at all. all doctrine bundle versions that are maintained contain this mappings pass, so we need no copy in core bundle. removing it, and adjusting the other bundles.
